src/internal/connector/graph_connector_onedrive_test.go

```diff
@@ -213,7 +213,7 @@ func (c *onedriveCollection) withFile(name string, fileData []byte, perm permDat
 func (c *onedriveCollection) withFolder(name string, perm permData) *onedriveCollection {
 	switch c.backupVersion {
```

Comment:
This switch looks like an opportunity for gotchas.  What about this instead?  This way we only need to update it in case of actual exceptions to the rule.
```go
if c.backupVersion > version.Backup || c.backupVersion < 0 {
  assert.FailNowf(...)
}

switch c.backupVersion {
case version.OneDrive1DataAndMetaFiles, 2, version.OneDrive3IsMetaMarker:
  // ...
default:
  return c
}
```

Reply:
this is one of the things I'm on the fence on. It's not necessarily a bad thing to make sure that people check this when bumping the version as they may have changed things that require updates to this logic